src/test/java/com/instana/operator/AgentDeployerTest.java

```diff
@@ -70,20 +78,75 @@ void daemonset_must_include_specified_image() {
   @Test
   void daemonset_must_include_image_from_csv_if_specified() {
     AgentDeployer deployer = new AgentDeployer();
+    deployer.setDefaultClient(client);
     deployer.setEnvironment(singleVar(RELATED_IMAGE_INSTANA_AGENT, "other/image:some-tag"));
 
-    InstanaAgent crd = new InstanaAgent();
-    crd.setSpec(new InstanaAgentSpec());
+    InstanaAgent customResource = new InstanaAgent();
+    customResource.setSpec(new InstanaAgentSpec());
 
     DaemonSet daemonSet = deployer.newDaemonSet(
-        crd,
+        customResource,
         client.inNamespace("instana-agent").apps().daemonSets());
 
     Container agentContainer = getAgentContainer(daemonSet);
 
     assertThat(agentContainer.getImage(), is("other/image:some-tag"));
   }
 
+  @Test
+  public void daemonset_must_include_version_label_if_specified_on_crd() {
+    // create crd from yaml
+    client.load(getClass().getResourceAsStream("/customresourcedefinition-with-version.yaml"))
+        .createOrReplace()
+        .stream()
+        .findAny()
+        .orElse(null);
+
+    AgentDeployer deployer = new AgentDeployer();
+    deployer.setDefaultClient(client);
+    deployer.setEnvironment(empty());
+
+    InstanaAgent customResource = new InstanaAgent();
+    customResource.setSpec(new InstanaAgentSpec());
+
+    DaemonSet daemonSet = deployer.newDaemonSet(
+        customResource,
+        client.inNamespace("instana-agent").apps().daemonSets());
+
+    Map<String, String> labels = daemonSet.getMetadata().getLabels();
+    assertThat(labels, hasEntry(is("app.kubernetes.io/managed-by"), is("instana-agent-operator")));
+    assertThat(labels, hasEntry(is("app.kubernetes.io/version"), is("0.3.8")));
+  }
+
+  @Test
+  public void daemonset_must_not_include_version_label_if_not_specified_on_crd() {
+    // create crd from yaml and remove version label
+    client
+        .load(getClass().getResourceAsStream("/customresourcedefinition-without-version.yaml"))
+        .createOrReplace()
+        .stream()
+        .findAny()
+        .orElse(null)
+        .getMetadata()
+        .getLabels()
+        .remove("app.kubernetes.io/version");
```

Comment:
Do you need to remove the version label if it is already not in the yaml? since this is `-without-version.yaml`?